### PR TITLE
Support SDL 2.0{4,5,6}

### PR DESCRIPTION
**Pull request by Octachron**
This PR adds support for the new functions and enums added between the version 2.04 and 2.06, with the exception of the hittest feature (`SDL_SetWindowHitTest` and friends) which is callback based, the vulkan symbol loader function and OS-specific hints.

Another point is that there is no link to the SDL wiki for most of the functions introduced in 2.06, since the SDL wiki has not been updated yet. More precisely, these links are removed in commit b08d1ec, so reverting the commit once the SDL wiki is fixed (or moved) might be an option .

I have updated  `test/test.ml` to test most of the new functions with the notable exception of the vulkan related functions since it would require too much setup to test in OCaml. However, I tested it with my vulkan binding ( https://github.com/Octachron/olivine/blob/master/examples/triangle.ml#151)  . If you wish, I could write the Vulkan glue code required in C to add a vulkan test to tsdl itself.

### CODE REVIEW

File: support/consts_stub.c [hunk 1]
```diff
@@ -63,13 +63,23 @@ void consts (FILE *fd)
   str_v (SDL_HINT_ORIENTATIONS);
   str_v (SDL_HINT_RENDER_DRIVER);
   str_v (SDL_HINT_RENDER_OPENGL_SHADERS);
+  str_v(SDL_HINT_RENDER_LOGICAL_SIZE_MODE);
```
Comment (dbuenzli) [inline support/consts_stub.c]: Style. Could you please add a space before the paren. See also below.

File: test/test.ml [hunk 2]
```diff
@@ -416,8 +437,11 @@ let test_renderers () =
           log " Render target supported: %b" (Sdl.render_target_supported r);
           let rect = Sdl.Rect.create 20 30 20 20 in
           assert (Sdl.render_set_clip_rect r (Some rect) = Ok ());
+          assert( Sdl.render_is_clip_enabled r);
```
Comment (dbuenzli) [inline test/test.ml]: Style. Could you please add a space before the paren and remove the one after it.

File: test/test.ml [hunk 3]
```diff
@@ -946,6 +993,7 @@ let test_mouse () =
       ignore (Sdl.show_cursor true);
       ignore (Sdl.get_cursor_shown ());
       Sdl.pump_events ();
+      assert ( Sdl.warp_mouse_global 50 50 = Ok () );
```
Comment (dbuenzli) [inline test/test.ml]: Style. Could you please remove the space before after the paren.

**Comment by Octachron** [comment]
I have fixed the style issues, and added my test for the `load_file` function which comes with a basic data file. I am not sure if it should stay. 

Concerning the wiki links, I removed them because I had no idea if the wiki will be updated any time soon, see https://bugzilla.libsdl.org/show_bug.cgi?id=3841.

**Comment by Octachron** [comment]
A small update: nearly two months and one version later, the upstream documentation has still not been updated (and obtaining write access on the SDL wiki is really bothersome).

**Comment by dbuenzli** [comment]
Thanks your patches have been merged.

**Pull request closed without merging**